card/lib/card/location.rb

```diff
@@ -0,0 +1,45 @@
+# -*- encoding : utf-8 -*-
+# helper for urls, links, redirects, and other location related things.
+#  note: i'm sure this isn't the optimal name..
+
+module Card::Location
+
+ # -----------( urls and redirects from application.rb) ----------------
+
+  # TESTME
+  def page_path title, opts={}
+    
+    format = opts[:format] ? ".#{opts.delete(:format)}"  : ''
+    query  = opts.present? ? "?#{opts.to_param}"         : ''
+    card_path "#{title.to_name.url_key}#{format}#{query}"
+  end
+
+  def card_path rel #should be in smartname?
+    rel_path = Card===rel ? rel.cardname.url_key : rel.to_s
+    if rel_path =~ /^\//
+      rel_path
+    else
+      "#{ Cardio.config.relative_url_root }/#{ rel_path }"
+    end
+  end
+
+  def card_url rel #should be in smartname?
+    if rel =~ /^https?\:/
+      rel
+    else
+      "#{ Card::Env[:protocol] }#{ Card::Env[:host] }#{ card_path rel }"
+    end
+  end
+
+
+  # Links ----------------------------------------------------------------------
+
+  def link_to_page( text, title=nil, options={})
+    title ||= text
+    url_options = {}
+    [:type, :view].each { |k| url_options[k] = options.delete(k) if options[k] }
+    url = page_path( title, url_options )
+    link_to text, url, options
+  end
+
```

Comment:
And I missed something ... there is one method used from the view context, link_to in the last method (link_to_path ?), and that is only used in a couple of places.  The others can all be class methods.

Now I'm not sure about why split off the session (history) methods, the session is just one more item in the view context.

Maybe they all go in Decko::Location (well not the 'class method' like ones) ?  I'll go with your direction on this.